financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/common/ConsentException.java

```diff
@@ -52,36 +53,50 @@ public ConsentException(ResponseStatus status, String errorMessage) {
                 errorMessage, null);
     }
 
+    public ConsentException(ResponseStatus status, JSONObject payload) {
+
+        this.status = status;
+        this.payload = payload;
+    }
+
     /**
      * This method is created to send error redirects in the authorization flow. The parameter validations are done
      * in compliance with the OAuth2 and OIDC specifications.
      *
-     * @param errorRedirectURI REQUIRED The base URI which the redirect should go to.
+     * @param errorURI         REQUIRED The base URI which the redirect should go to.
      * @param error            REQUIRED The error code of the error. Should be a supported value in OAuth2/OIDC
      * @param errorDescription OPTIONAL The description of the error.
      * @param state            REQUIRED if a "state" parameter was present in the client authorization request.
      */
-    public ConsentException(URI errorRedirectURI, AuthErrorCode error, String errorDescription, String state) {
+    public ConsentException(URI errorURI, AuthErrorCode error, String errorDescription, String state) {
 
-        if (errorRedirectURI != null && error != null) {
+        if (errorURI != null && error != null) {
             //add 302 as error code since this will be a redirect
+            errorRedirectURI = errorURI;
             this.status = ResponseStatus.FOUND;
-            this.payload = createDefaultErrorObject(errorRedirectURI, error.toString(), errorDescription, state);
+            this.payload = createDefaultErrorObject(errorURI, error.toString(), errorDescription, state);
         }
     }
 
     public JSONObject createDefaultErrorObject(URI redirectURI, String errorCode, String errorMessage, String state) {
```

Comment:
```suggestion
    public JSONObject createDefaultErrorObject(URI redirectURI, String errorCode, String errorDescription, String state) {
```

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/130/commits/1e13ec5af065388311c13d8ca7f04ef804d84184